### PR TITLE
[DAT-7850] - Diff & Snapshot test implementation

### PR DESCRIPTION
**Pull request by AlexanderSashchenko**

### CODE REVIEW

File: src/main/groovy/liquibase/harness/diff/DiffCommandTestHelper.groovy [hunk 1]
```diff
@@ -88,99 +55,84 @@ class DiffCommandTestHelper {
...
+    /** This method creates diffToCompare without JSONObjects referenced to databasechangelog* tables
+     because --excludeObjects flag doesn't work with --format=json extension for 'diff' command
+     */
+    static JSONObject createDiffToCompare(OutputStream diffOutput) {
+        def generatedDiff = new JSONObject(diffOutput.toString().replaceAll("!!int", ""))
+                .getJSONObject("diff")// Replacement because of a bug in diff --format=json generation
+        def diffToCompare = new JSONObject()
+        def arrays = ["missingObjects", "unexpectedObjects", "changedObjects"]
+        def objects = ["missingObject", "unexpectedObject", "changedObject"]
+        for (int i = 0; i < 3; i++) {
```
Comment (KushnirykOleh) [inline src/main/groovy/liquibase/harness/diff/DiffCommandTestHelper.groovy:98]: magic number, use array length instead

File: src/main/groovy/liquibase/harness/util/TestUtils.groovy [hunk 2]
```diff
@@ -12,33 +10,6 @@ import java.util.logging.Logger
 
 class TestUtils {
 
-    static SortedMap<String, String> resolveInputFilePaths(DatabaseUnderTest database, String basePath, String inputFormat) {
```
Comment (KushnirykOleh) [inline src/main/groovy/liquibase/harness/util/TestUtils.groovy:15]: This method was covered by TestUtilsTest.getInputFilesPaths test. Now this test is failing (but we don't see it in build)

**Review by KushnirykOleh — CHANGES_REQUESTED** [review]

File: src/main/groovy/liquibase/harness/diff/DiffCommandTestHelper.groovy (hunk 1 above)
Comment (AlexanderSashchenko) [reply src/main/groovy/liquibase/harness/diff/DiffCommandTestHelper.groovy:98]: ++

File: src/main/groovy/liquibase/harness/util/TestUtils.groovy (hunk 2 above)
Comment (AlexanderSashchenko) [reply src/main/groovy/liquibase/harness/util/TestUtils.groovy:15]: ++ moved to file utils

File: src/main/groovy/liquibase/harness/change/ChangeObjectTests.groovy [hunk 3]
```diff
@@ -2,24 +2,25 @@ package liquibase.harness.change
...
     @Unroll
-    def "apply #testInput.changeObject against #testInput.databaseName #testInput.version; verify generated SQL and DB snapshot"() {
+    def "apply #testInput.changeObject against #testInput.databaseName #testInput.version"() {
         given: "read expected sql and snapshot files, create arguments map for executing command scope"
-        String expectedSql = parseQuery(FileUtils.getExpectedSqlFileContent(testInput.changeObject, testInput.databaseName,
-                testInput.version, "liquibase/harness/change/expectedSql"))
-        String expectedSnapshot = FileUtils.getExpectedJSONFileContent(testInput.changeObject, testInput.databaseName,
-                testInput.version, "liquibase/harness/change/expectedSnapshot")
-        Map<String, Object> argsMap = new HashMap<>()
+        def expectedSql = parseQuery(getSqlFileContent(testInput.changeObject, testInput.databaseName, testInput.version,
```
[the remark this answers is not in the document]
Comment (AlexanderSashchenko) [reply src/main/groovy/liquibase/harness/change/ChangeObjectTests.groovy]: ++

**Review by KushnirykOleh — APPROVED** [review]

**Pull request merged**